pkg/database/user.go

```diff
@@ -35,6 +35,11 @@ type User struct {
 	AdminRealms     []*Realm `gorm:"many2many:admin_realms"`
 }
 
+// CacheKey returns the key for this user in the distributed cache.
+func (u *User) CacheKey() string {
+	return fmt.Sprintf("users:by_email:%s", u.Email)
```

Comment:
hmm... we cache users by id and email (I think)

Reply:
where this matters is in middleware/auth.go - line 77ish.
there it is cached by email

it's also interesting that the callback doesn't hit this

Reply:
Oh hmmm... [it does](https://github.com/google/exposure-notifications-verification-server/blob/main/pkg/database/database.go#L195)? Were you seeing something else? Which user query were you running?

Reply:
the callbacks work - don't know what I Was doing wrong before